### PR TITLE
configuration_manual/authentication: Remove shadow passdb driver

### PR DESCRIPTION
**Pull request by fleischie**
- Mark shadow passdb driver as deprecated, and
- remove unnecessary references to removed auth driver.

### CODE REVIEW

File: source/admin_manual/system_users_used_by_dovecot.rst [hunk 1]
```diff
@@ -137,9 +137,9 @@ default is to use root, because it's guaranteed to have access to all the
 password databases. If you don't need this, you should change it to
 ``$default_internal_user``.
 
-:ref:`authentication-pam` and :ref:`authentication-shadow` passdbs are usually
-configured to read ``/etc/shadow`` file. Even this doesn't need root access if
-the file is readable by shadow group:
+:ref:`authentication-pam` passdb is usually configured to read ``/etc/shadow``
```
Comment (cmouse) [inline source/admin_manual/system_users_used_by_dovecot.rst:140]: this is actually bit misleading. pam is usually configured to read /etc/shadow, not passdb. it was wrong in the original text as well.

File: source/configuration_manual/authentication/password_schemes.rst [hunk 2]
```diff
@@ -104,7 +104,7 @@ setting in ``dovecot-sql.conf.ext``
 :ref:`authentication-passwd_file` : CRYPT is used
 by default, but can be changed with ``scheme`` parameter in passdb args.
 
-:ref:`authentication-passwd`, :ref:`authentication-shadow`, ``vPopMail``: CRYPT is used by default
+:ref:`authentication-passwd`, ``vPopMail``: CRYPT is used by default
```
Comment (cmouse) [inline source/configuration_manual/authentication/password_schemes.rst:107]: vpopmail has already been removed in 2.3.13

**Review by cmouse — CHANGES_REQUESTED** [review]
can you add separate commit to remove any vpopmail / vPopMail references, please?

File: source/configuration_manual/authentication/password_schemes.rst (hunk 2 above)
Comment (fleischie) [reply source/configuration_manual/authentication/password_schemes.rst]: I removed all vpopmail references in a separate commit.

**Review by cmouse — APPROVED** [review]

**Pull request merged**